hypothesis-python/RELEASE.rst

```diff
@@ -0,0 +1,4 @@
+RELEASE_TYPE: minor
+
+This introduces the rewriting of length filters in ``text`` strategies, 
+utilizing ``partial`` and ``lambda`` functions for efficient handling of length constraints.
```

Comment:
Patch release since this doesn't change any public interfaces and may or may not work in any particular case.  We'll also want to use cross-reference syntax for `text`  and also a link to the relevant issue.  Finally, I'd love to have a "Thanks to Reagan Lee for implementing this feature!" note at the end 🙂